client/generator/src/main/java/io/openlineage/client/JavaPoetGenerator.java

```diff
@@ -0,0 +1,174 @@
+package io.openlineage.client;
+
+import static io.openlineage.client.TypeResolver.titleCase;
+import static javax.lang.model.element.Modifier.FINAL;
+import static javax.lang.model.element.Modifier.PRIVATE;
+import static javax.lang.model.element.Modifier.PUBLIC;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.lang.model.element.Modifier;
+
+import com.fasterxml.jackson.annotation.JsonAnyGetter;
+import com.fasterxml.jackson.annotation.JsonAnySetter;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.squareup.javapoet.AnnotationSpec;
+import com.squareup.javapoet.ClassName;
+import com.squareup.javapoet.CodeBlock;
+import com.squareup.javapoet.FieldSpec;
+import com.squareup.javapoet.JavaFile;
+import com.squareup.javapoet.MethodSpec;
+import com.squareup.javapoet.MethodSpec.Builder;
+import com.squareup.javapoet.ParameterSpec;
+import com.squareup.javapoet.ParameterizedTypeName;
+import com.squareup.javapoet.TypeName;
+import com.squareup.javapoet.TypeSpec;
+
+import io.openlineage.client.TypeResolver.ArrayType;
+import io.openlineage.client.TypeResolver.Field;
+import io.openlineage.client.TypeResolver.ObjectType;
+import io.openlineage.client.TypeResolver.PrimitiveType;
+import io.openlineage.client.TypeResolver.Type;
+import io.openlineage.client.TypeResolver.TypeVisitor;
+
+/**
+ * Generates a JavaClass with all the types as inner classes
+ */
+public class JavaPoetGenerator {
+
+  private static final String PACKAGE = "io.openlineage.client";
+  private static final String CONTAINER_CLASS_NAME = "OpenLineage";
```

Comment:
_Minor_: Would we want to name the class `OpenLineageClient`?

Reply:
I see, this is generating the `OpenLineage` model with the nested internal class. Is the idea to also generate `OpenLineageClient`?

Reply:
I think the Client would not be generated and be a class in the repo. This is the first step.